code/modules/vehicles/mecha/mech_melee_attack.dm

```diff
@@ -28,10 +31,31 @@
 		playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
 	else
 		add_dent(WALL_DENT_HIT)
+	TIMER_COOLDOWN_START(mecha_attacker, COOLDOWN_MECHA_MELEE_ATTACK, mecha_attack_cooldown)
 	..()
 	return 100 //this is an arbitrary "damage" number since the actual damage is rng dismantle
 
-/obj/mech_melee_attack(obj/vehicle/sealed/mecha/mecha_attacker, mob/living/user)
+/obj/structure/mech_melee_attack(obj/vehicle/sealed/mecha/mecha_attacker, mob/living/user, mecha_attack_cooldown)
+	if(!user.combat_mode)
+		return 0
+
+	mecha_attacker.do_attack_animation(src)
+	switch(mecha_attacker.damtype)
+		if(BRUTE)
+			playsound(src, 'sound/weapons/punch4.ogg', 50, TRUE)
+		if(BURN)
+			playsound(src, 'sound/items/welder.ogg', 50, TRUE)
+		else
+			return 0
+	mecha_attacker.visible_message(span_danger("[mecha_attacker] hits [src]!"), span_danger("You hit [src]!"), null, COMBAT_MESSAGE_RANGE)
+	TIMER_COOLDOWN_START(mecha_attacker, COOLDOWN_MECHA_MELEE_ATTACK, mecha_attack_cooldown)
```

Comment:
I think putting the onus on the target procs to start the cooldowns is a bit stinky. This stuff can easily be forgotten or accidentally misplaced

This proc seems to have a return value but a quick glance says that return value is not read anywhere, why not make it so we start the cooldown if this proc returns a value or something?

Reply:
Lemme see what I can do.

Reply:
Was easy enough.